app/scripts/metamask-controller.test.js

```diff
@@ -193,6 +196,13 @@ describe('MetaMaskController', function () {
     await ganacheServer.quit();
   });
 
+  describe('should reset state on first time profile load', function () {
+    it('should reset state', function () {
+      assert(metamaskController.resetStates.calledOnce);
+      assert.equal(globalThis.isFirstTimeProfileLoaded, false);
+    });
+  });
+
```

Comment:
Nit: will be nice to have other test also that reset does not run if `globalThis.isFirstTimeProfileLoaded` is false.